testsuite/features/step_definitions/api_common.rb

```diff
@@ -245,21 +245,26 @@
   # Get the list of child channels for this base channel
   child_channels = $api_test.channel.software.list_child_channels(base_channel)
 
-  # Select all the child channels for this client
-  client.sub! 'ssh_minion', 'minion'
-  if client.include? 'buildhost'
-    selected_child_channels = ["custom_channel_#{client.sub('buildhost', 'minion')}", "custom_channel_#{client.sub('buildhost', 'client')}"]
-  elsif client.include? 'terminal'
-    selected_child_channels = ["custom_channel_#{client.sub('terminal', 'minion')}", "custom_channel_#{client.sub('terminal', 'client')}"]
-  else
-    custom_channel = "custom_channel_#{client}"
-    selected_child_channels = [custom_channel]
-  end
-  child_channels.each do |child_channel|
-    selected_child_channels.push(child_channel) unless child_channel.include? 'custom_channel'
-  end
+  # Filter out the custom channels
+  # This is needed because we might have both a traditional custom channel and a Salt custom channel
+  child_channels.reject! { |channel| channel.include? 'custom_channel' }
 
-  $api_test.activationkey.add_child_channels(key, selected_child_channels)
+  # Re-add the desired custom channel
+  client.sub! 'ssh_minion', 'minion'
+  client.sub! 'buildhost', 'minion'
+  client.sub! 'terminal', 'minion'
+  client.sub! 'monitoring_server', 'sle15sp4_minion'
+  custom_channel = if client.include? 'rocky8'
+                     "no-appstream-8-result-custom_channel_#{client}"
+                   elsif client.include? 'rocky9'
+                     "no-appstream-9-result-custom_channel_#{client}"
+                   else
+                     "custom_channel_#{client}"
+                   end
+  child_channels.push(custom_channel)
```

Comment:
What happen if the client has multiple custom channels?

Reply:
It does not happen, but if you really want to ignore YAGNI, I can make it an array again.

Reply:
I see multiple custom channels defined in our JSON file for each client.
https://github.com/SUSE/spacewalk/issues/19518#issuecomment-1329379893

Reply:
I said nothing Eric. We only have one custom channel per client, and we have inside that one all our MU repos.

Reply:
Perfect then